azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/extensions/AzureComputeImageExtension.java

```diff
@@ -118,26 +121,47 @@ public Image call() throws Exception {
                            disks[1] = datadiskObject.get("name");
 
                            VirtualMachine vm = api.getVirtualMachineApi(group).get(id);
-                           String location = vm.location();
-                           final VMImage ref = VMImage.create(CUSTOM_IMAGE_PREFIX + group, CUSTOM_IMAGE_PREFIX + name, disks[0], disks[1], location, false);
+                           final VMImage ref = VMImage.create(group, storageAccountName, disks[0], disks[1], name, "custom", vm.location());
                            return imageReferenceToImage.apply(ref);
                         }
                      }
                   }
                }
                throw new UncheckedTimeoutException("Image was not created within the time limit: "
-                       + cloneTemplate.getName());
+                     + cloneTemplate.getName());
             }
          });
       } else {
          final String illegalStateExceptionMessage = format("Node %s was not suspended within %sms.",
-                 id, azureComputeConstants.operationTimeout());
+               id, azureComputeConstants.operationTimeout());
          throw new IllegalStateException(illegalStateExceptionMessage);
       }
    }
 
    @Override
    public boolean deleteImage(String id) {
+
+      String[] fields = VMImageToImage.decodeFieldsFromUniqueId(id);
+      boolean custom = fields.length == 6 ? true : false;
+      if (custom) {
+         /* id fields indexes
+         0: imageReference.location) + "/" +
+         1: imageReference.group + "/" +
+         2: imageReference.storage + "/" +
+         3: imageReference.vhd1 + "/" +
+         4: imageReference.offer + "/" +
+         5: imageReference.location;
+         */
+         String group = fields[1];
+         String storage = fields[2];
+         StorageServiceKeys keys = api.getStorageAccountApi(group).getKeys(storage);
+
+         // This removes now all the images in this storage. At least in theory, there should be just one and if there is
+         // more, they should be copies of each other.
+         BlobHelper.deleteContainerIfExists(storage, keys.key1(), "system");
```

Comment:
Is it possible to have `deleteContainerIfExists` return some type of status here to verify that the container has been deleted successfully?